pkg/arvo/sys/zuse.hoon

```diff
@@ -7950,6 +7950,60 @@
     ?:  (compare key.n.a key.n.b)
       $(l.b $(b l.b, r.a ~), a r.a)
     $(r.b $(b r.b, l.a ~), a l.a)
+  ::
+  ::  +subset: take a range excluding start and/or end and all elements
+  ::  outside the range
+  ::
+  ++  subset
+    |=  $:  tre=(tree item)
+            start=(unit key)
+            end=(unit key)
+        ==
+    ^-  (tree item)
+    |^
+    ?:  ?&(?=(~ start) ?=(~ end))
+      tre
+    ?~  start
+      (del-span tre %end end)
+    ?~  end
+      (del-span tre %start start)
+    ?>  (lth u.start u.end)
+    =.  tre  (del-span tre %start start)
+    (del-span tre %end end)
+    ::
+    ++  del-span
+      |=  [a=(tree item) b=?(%start %end) c=(unit key)]
+      ^-  (tree item)
+      ?~  a  a
+      ?~  c  a
+      ?-  b
+          %start
+        ::  found key
+        ?:  =(key.n.a u.c)
+          (nip a(l ~))
+        ::  traverse to find key
+        ?:  (compare key.n.a u.c)
+          ::  found key to the left of start
+          (del-span (nip a(l ~)) b c)
+        ::  found key to the right of start
+        =/  lef  $(a l.a)
+        =.  l.a  lef
+        a
```

Comment:
Feels a little strange to have two ways of recursing, would probably be cleaner to just choose one.  My preference would be the `$` recursion.

Also I feel these few lines would be cleaner as `a(l $(a l.a))` or at least `=.  l.a  $(a l.a)  a`

Reply:
I'll clean up the recursion and look at your other comments.

Reply:
Done.